04_testing/severity_analysis.R

```diff
@@ -964,3 +965,396 @@ write_csv(
+      yend = c(1:4 + 0.5, 1:4 + 0.5)
+    ),
+    aes(
+      x = x,
+      xend = xend,
+      y = y,
+      yend = yend
+    ),
+    color = "#18998F"
+  ) +
+  geom_segment(
+    data = data.frame(
+      x = c(0:3 + 0.5, 1:4 + 0.5),
+      xend = c(1:4 + 0.5, 1:4 + 0.5),
+      y = c(1:4 + 0.5, 1:4 + 0.5),
+      yend = c(1:4 + 0.5, 2:5 + 0.5)
+    ),
+    aes(
+      x = x,
+      xend = xend,
+      y = y,
+      yend = yend
+    ),
+    color = "#007CE0"
+  ) +
+  geom_text(
+    data = data.frame(
+      sector = "Protection (HLP)",
+      severity = c(4.5, 2),
+      intersectoral = c(2, 4.5),
+      label = c(
+        "Sectoral severity higher\nthan intersectoral",
+        "Sectoral severity\nlower than\nintersectoral"
+      )
+    ),
+    aes(
+      label = label
+    ),
+    family = "Roboto",
+    size = 2.5
+  ) +
+  facet_wrap(
+    ~sector
+  ) +
+  scale_fill_gradient(
+    low = "white",
+    high = "#1EBFB3",
+    na.value = "white",
+    labels = scales::percent_format(1)
+  ) +
+  labs(
+    y = "Sectoral severity",
+    title = "Comparison of sectoral and intersectoral severity",
+    x = "Intersectoral severity",
+    fill = "% of areas"
+  ) +
+  theme_minimal() +
+  theme(
+    plot.title = element_text(
+      face = "bold",
+      size = 22,
+      margin = margin(10, 10, 10, 10, "pt"),
+      family = "Roboto",
+      hjust = .5
+    ),
+    plot.background = element_rect(
+      fill = "white"
+    ),
+    axis.text = element_text(
+      face = "bold",
+      size = 10,
+      family = "Roboto"
+    ),
+    axis.title.y = element_text(
+      face = "bold",
+      size = 12,
+      family = "Roboto",
+      margin = margin(r = 20)
+    ),
+    axis.title.x = element_text(
+      face = "bold",
+      size = 12,
+      family = "Roboto",
+      margin = margin(t = 20)
+    ),
+    legend.text = element_text(
+      size = 12,
+      family = "Roboto"
+    ),
+    legend.key.width = unit(1, "cm"),
+    legend.position = "bottom",
+    panel.grid.minor = element_blank(),
+    legend.background = element_rect(fill = "transparent"),
+    legend.box.background = element_rect(fill = "transparent"),
+    strip.text = element_text(
+      size = 16,
+      family = "Roboto"
+    )
+  )
+
+ggsave(
+  file.path(
+    file_paths$output_dir_sev,
+    "graphs",
+    "2022_hno_compr_sector_is_tiles.png"
+  ),
+  width = 10,
+  height = 8,
+  units = "in"
+)
+
+write_csv(
+  df_sect_is_comp,
+  file.path(
+    file_paths$output_dir_sev,
+    "graphs",
+    "datasets",
+    "2022_hno_compr_sector_is_tiles.csv"
+  )
+)
+
+##################################
+#### CORRELATIONS BETWEEN ALL ####
+##################################
+
+df_corr <- df %>%
+  select(
+    adm0_name,
+    disaggregation,
+    sector,
+    severity
+  ) %>%
+  filter(
+    sector != "Intersectoral (raw)"
+  ) %>%
+  pivot_wider(
+    names_from = sector,
+    values_from = severity
+  ) %>%
+  select(
+    Intersectoral:`Protection (CP)`
```

Comment:
in my machine the last column after pivot_wider is Protection (HLP) and running this script misses 4 sectors including Health, CCCM, MA and HLP